src/Commands/Admin/AddHubToHubAssociation.cs

```diff
@@ -33,22 +33,57 @@ protected override void ExecuteCmdlet()
         {
             if (ParameterSetName == ParamSet_ById)
             {
-                HubSiteProperties sourceProperties = this.Tenant.GetHubSitePropertiesById(Source);
+                HubSiteProperties sourceProperties = Tenant.GetHubSitePropertiesById(Source);
                 ClientContext.Load(sourceProperties);
                 sourceProperties.ParentHubSiteId = Target;
                 sourceProperties.Update();
                 ClientContext.ExecuteQueryRetry();
             }
             else
             {
-                HubSiteProperties sourceProperties = this.Tenant.GetHubSitePropertiesByUrl(SourceUrl);
-                ClientContext.Load(sourceProperties);
-                var targetSite = this.Tenant.GetSiteByUrl(TargetUrl);
-                ClientContext.Load(targetSite);
-                ClientContext.ExecuteQueryRetry();
-                sourceProperties.ParentHubSiteId = targetSite.HubSiteId;
-                sourceProperties.Update();
+                ClientObjectList<HubSiteProperties> hubSitesProperties = Tenant.GetHubSitesProperties();
```

Comment:
Consider changing to checking the HubSiteId property on the tenantsite object instead of retrieving all hub sites.

Reply:
Fixed this , thanks for spotting it